distributed/comm/tcp.py

```diff
@@ -299,16 +299,25 @@ async def write(self, msg, serializers=None, on_error="message"):
             # trick to enque all frames for writing beforehand
             for each_frame_nbytes, each_frame in zip(frames_nbytes, frames):
                 if each_frame_nbytes:
-                    if stream._write_buffer is None:
-                        raise StreamClosedError()
+                    each_frame = memoryview(each_frame)
 
                     if isinstance(each_frame, memoryview):
                         # Make sure that `len(data) == data.nbytes`
                         # See <https://github.com/tornadoweb/tornado/pull/2996>
                         each_frame = ensure_memoryview(each_frame)
 
-                    stream._write_buffer.append(each_frame)
-                    stream._total_write_index += each_frame_nbytes
+                    # Workaround for OpenSSL 1.0.2 (can drop with OpenSSL 1.1.1)
+                    for i, j in sliding_window(
+                        2, range(0, each_frame_nbytes + C_INT_MAX, C_INT_MAX)
```

Comment:
The constant was since renamed for clarity

```suggestion
                        2, range(0, each_frame_nbytes + OPENSSL_MAX_CHUNKSIZE, OPENSSL_MAX_CHUNKSIZE)
```